tensorflow/compiler/xla/service/hlo_instruction.cc

```diff
@@ -3823,6 +3828,24 @@ StatusOr<PrecisionConfig::Precision> StringToPrecision(const string& name) {
   return found->second;
 }
 
+StatusOr<CustomCallSchedule> StringToCustomCallSchedule(const string& name) {
+  static std::unordered_map<string, CustomCallSchedule>* map = [] {
+    static auto* map = new std::unordered_map<string, CustomCallSchedule>;
+    for (int i = 0; i < CustomCallSchedule_ARRAYSIZE; i++) {
+      if (CustomCallSchedule_IsValid(i)) {
+        auto value = static_cast<CustomCallSchedule>(i);
+        (*map)[CustomCallScheduleToString(value)] = value;
+      }
+    }
+    return map;
+  }();
+  auto found = map->find(absl::AsciiStrToLower(name));
```

Comment:
This looks not thread safe. Do we need a mutex? (Cf. global map examples in XLA:GPU, e.g. autotuning)

Reply:
Static local variable initialization is thread safe ([ref to Stackoverflow](https://stackoverflow.com/questions/8102125/is-local-static-variable-initialization-thread-safe-in-c11)).

So, if we don't update the map, this piece of code is thread safe. Let me see if I can update the declaration to `static const auto* map`.

Honestly I just used the coding style in the same file ([here](https://github.com/tensorflow/tensorflow/pull/48806/files#diff-9c21cb9f1d5f87998d7ea54e5f659af355f4d3096f9ba94871f32c112087f75aR3812)).